File: set_admin_claims.js
// Script para asignar claims de superadministrador y administrador de ciudad en Firebase Auth
// Ejecutar con: node set_admin_claims.js

const admin = require('firebase-admin');
const serviceAccount = require('./serviceAccountKey.json'); // Debes descargar tu clave de servicio desde Firebase Console

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount)
});

async function setClaims() {
  // Superadministrador global
  await admin.auth().setCustomUserClaims(
    (await admin.auth().getUserByEmail('admin@trafico-map.com')).uid,
    { role: 'superadmin' }
  );
  console.log('✅ Claim superadmin asignado a admin@trafico-map.com');

  // Administrador de La Plata
  await admin.auth().setCustomUserClaims(
    (await admin.auth().getUserByEmail('admin@laplata.com')).uid,
    { role: 'admin', city: 'laplata' }
  );
  console.log('✅ Claim admin/city asignado a admin@laplata.com');

  // Ejemplo: agregar más administradores de ciudad aquí
  // await admin.auth().setCustomUserClaims(
  //   (await admin.auth().getUserByEmail('admin@cordoba.com')).uid,
  //   { role: 'admin', city: 'cordoba' }
  // );
  // console.log('✅ Claim admin/city asignado a admin@cordoba.com');

  process.exit(0);
}

setClaims().catch(e => {
  console.error('❌ Error:', e);
  process.exit(1);
});

File: set_admin_claims.py
#!/usr/bin/env python3
import firebase_admin
from firebase_admin import credentials, auth
import os


cred_path = 'service-account-key.json'
if not os.path.exists(cred_path):
    print("❌ No encontré service-account-key.json")
    exit(1)

cred = credentials.Certificate(cred_path)
firebase_admin.initialize_app(cred)

admin_email = "admin@laplata.com"

try:
    user = auth.get_user_by_email(admin_email)
    print(f"✅ Usuario encontrado: {user.email}")
    new_claims = {
        'admin': True,
        'role': 'admin',
        'rol': 'admin',
        'email': admin_email
    }
    auth.set_custom_user_claims(user.uid, new_claims)
    print("✅ Custom claims actualizados exitosamente")
except Exception as e:
    print(f"❌ Error: {e}")
    exit(1)
